simple_test.sh

```diff
@@ -1,48 +1,81 @@
+worker_cnt=$2
+test_file=$3
+
+# Initalize integers to be used later
+supervisor_machine=5001
+worker_machine=5001
+window=0
+pane=0
+
 tmux has-session -t stest
 
 if [ $? != 0 ]
-then
-    # Tmux window with 4 panes
-    # One pane will be the supervisor on stu (test:0)
-    tmux new-session -s stest -n 'simple-test' -d
-    tmux split-window -h -t stest:0.0
-    tmux send-keys -t stest:0.0 "cd $PROJECT_DIRECTORY" Enter
-    tmux send-keys -t stest:0.0 'cd cmd/supervisor' Enter
-    tmux send-keys -t stest:0.0 'go build' Enter
-    tmux send-keys -t stest:0.0 './supervisor :5001' Enter
-    # Create Worker 1 (test:1) pane
-    tmux split-window -v -t stest:0.1
-    # Startup Worker 1
-    tmux send-keys -t stest:0.1 'ssh -o StrictHostKeyChecking=no l25001.cs.jmu.edu' Enter
-    tmux send-keys -t stest:0.1 "cd $PROJECT_DIRECTORY" Enter
-    tmux send-keys -t stest:0.1 'cd cmd/worker' Enter
-    tmux send-keys -t stest:0.1 'go build' Enter
-    tmux send-keys -t stest:0.1 'sleep 1' Enter
-    tmux send-keys -t stest:0.1 './worker http://stu.cs.jmu.edu:5001 5002' Enter
-    # Create Worker 2 (test:2)
-    tmux split-window -h -t stest:0.2
-    # Startup Worker 2
-    tmux send-keys -t stest:0.2 'ssh -o StrictHostKeyChecking=no l25002.cs.jmu.edu' Enter
-    tmux send-keys -t stest:0.2 "cd $PROJECT_DIRECTORY" Enter
-    tmux send-keys -t stest:0.2 'cd cmd/worker' Enter
-    tmux send-keys -t stest:0.2 'go build' Enter
-    tmux send-keys -t stest:0.2 'sleep 1' Enter
-    tmux send-keys -t stest:0.2 './worker http://stu.cs.jmu.edu:5001 5003' Enter
-    # Create Client (test:3) pane
-    tmux select-layout tiled
-    # Lastly, switch to Client pane
-    tmux send-keys -t stest:0.3 'ssh -o StrictHostKeyChecking=no l25009.cs.jmu.edu' Enter
-    tmux send-keys -t stest:0.3 "cd $PROJECT_DIRECTORY" Enter
-    tmux send-keys -t stest:0.3 'cd cmd/client' Enter
-    tmux send-keys -t stest:0.3 'go build' Enter
-    tmux send-keys -t stest:0.3 'sleep 2' Enter
-    tmux send-keys -t stest:0.3 "./client http://stu.cs.jmu.edu:5001 $PROJECT_DIRECTORY/test/hello.sh" Enter
-    # Switch back to supervisor
-    tmux select-pane -t stest:0.0
+  then
+    # Tmux window with worker_cnt worker panes, a supervisor pane, and a client pane
+    # One pane will be the supervisor on stu (stest:0.0)
+    tmux new-session -s stest -n "simple-test" -d
+    tmux split-window -h -t stest:$window.$pane
+    tmux send-keys -t stest:$window.$pane "cd $PROJECT_DIRECTORY" Enter
+    tmux send-keys -t stest:$window.$pane "cd cmd/supervisor" Enter
+    tmux send-keys -t stest:$window.$pane "go build" Enter
+    tmux send-keys -t stest:$window.$pane "./supervisor :$supervisor_machine" Enter
+    # Setup Worker panes
+    for i in $(seq 1 $worker_cnt)
+      do
+    	# Create Worker i (stest:window.pane) pane
+	let "pane=i%20"
+        tmux split-window -v -t stest:$window.$pane
+	# Worker i will corespond to machine (5000 + i+1), this is offset by 1 with supervisor starting on 5001
```

Comment:
This might be a little pedantic, but the lab machine numbering system is

`l<lab number><machine number>`, so l25001 is computer 1 of lab 250

So starting counting at 5001 works, but it doesn't make a ton of sense with this numbering scheme in mind

Reply:
Oh gotcha, that makes a lot more sense. I think it would be good to have a lab # command-line argument as well. I can make this adjustment. Do the machine numbers for each lab start on 1? I was thinking it would be nice if I could have the supervisor start on machine 0 if there is one.

Reply:
There is no machine 0. We could have the supervisor just be stu.